src/Engine/Data/ShaderProgram.hpp

```diff
@@ -47,6 +47,8 @@ class RA_ENGINE_API ShaderProgram final
     /// Uniform setters
     template <typename T>
     void setUniform( const char* name, const T& value ) const;
+    template <typename T, typename E>
```

Comment:
What is `E`?

Reply:
maybe something I should have deleted, from by test.
If you have any guess how to specialize with enable_if and same_type to have only conversion if T is std::vector< Scalar>  and Scalar is double

Reply:
maybe simply specialize for std::vector<double> the typename replacement do the rest ...

Reply:
We need to check, but a simple `if (same_type )` should be enough: it is a constexpr that will be known at compile time, and thus probably optimized.

Reply:
It’s not sufficient since both branch of the if have to be valid code for the template parameters. And we want to make a simple copy which is not valid if Scalar is not the same as GL_SCALAR_PLAIN

Reply:
Hum right,
Then we can have a local template function with a unique template parameter, a boolean (computed with same_type), which is specialized with the two behaviors.